setup.py

```diff
@@ -226,6 +256,69 @@ def run(self):
         print("Unsupported command for openshift-ansible")
 
 
+class MoleculeTests(Command):
+    ''' Command for running molecule tests '''
+    description = "Run molecule tests"
+    user_options = [
+        ('roles=', 'r', 'which roles to test'),
+        ('excludes=', 'e', 'patterns to exclude'),
+    ]
+
+    def initialize_options(self):
+        ''' initialize_options '''
+        # pylint: disable=attribute-defined-outside-init
+        self.roles = None
+        self.excludes = None
+
+    def finalize_options(self):
+        ''' finalize_options '''
+        # pylint: disable=attribute-defined-outside-init
+        if isinstance(self.roles, string_types):
+            self.roles = self.roles.split(',')
+
+        if isinstance(self.excludes, string_types):
+            self.excludes = self.excludes.split(',')
+
+    def run(self):
+        ''' run command '''
+        if self.roles is not None:
+            print("Roles:\n{0}".format(yaml.dump(self.roles, default_flow_style=False)))
+        if self.excludes is not None:
+            print("Excludes:\n{0}".format(yaml.dump(self.excludes, default_flow_style=False)))
+
+        starting_dir = '.'
+        if self.roles is not None:
+            starting_dir = 'roles'
+
+        molecule_dirs = find_dirs(starting_dir, self.excludes, self.roles, 'molecule')
+
+        print("Found:\n{0}".format(yaml.dump(molecule_dirs, default_flow_style=False)))
+        base_dir = os.getcwd()
+
+        errors = ""
+        warnings = ""
+
+        for role in molecule_dirs:
```

Comment:
One of the reasons #2951 introduced Go to the repo is that it makes it so much easier and reliable to parallelize this loop of slow tests with clean reporting of errors, all based in Go's stdlib and `go test`.